modules/pel/peltool/comp_id.py

```diff
@@ -40,10 +47,11 @@ def getDisplayCompID(componentID: int, creatorID: str) -> str:
 
     # try the comp IDs file named after the creator ID
     if creatorID not in componentIDs:
-        compIDFile = getCompIDFilePath(creatorID)
-        if os.path.exists(compIDFile):
-            with open(compIDFile, 'r') as file:
-                componentIDs[creatorID] = json.load(file)
+        if checkCompIDFilePath[0]:
```

Comment:
I'm having a bit of a hard time following if this code assumes there is just 1 creator ID value per PEL...

I mention that because in hostboot PELs the BMC fills in an Extended User Data class with a BMC creator ID in a PEL where all the other creator IDs are hostboot.  Here it is alongside another section:
```
},
"Extended User Header": {
    "Section Version":          "1",
    "Sub-section type":         "0",
    "Created by":               "hostboot: htmgt",
    "Reporting Machine Type":   "9028-21B",
    "Reporting Serial Number":  "B000002",
    "FW Released Ver":          "NL1060_072",
    "FW SubSys Version":        "v2.7-10.720",
    "Common Ref Time":          "00/00/0000 00:00:00",
    "Symptom Id Len":           "76",
    "Symptom Id":               "BC8A26B9_000000E000000A00000000000020000000000000000000000000000000000000"
},
"Extended User Data": {
    "Section Version": "1",
    "Sub-section type": "1",
    "Created by": "bmc error logging",
    "BMCState": "Ready",
    "BootState": "SystemInitComplete",
    "ChassisState": "On",
    "FW Version ID": "fw1060.20-23-1060.2438.20240927a (NL1060_072)",
    "HostState": "Running",
    "System IM": "50004000"
},
```

Reply:
+1, I am also feeling difficulty while looking this code.
 
I'm not sure why we're overcomplicating this. After closely reviewing the flow, it seems that we're simply trying to retrieve the string value of the component ID from a component ID JSON file using the creator ID.

Why not just find all the creator component ID JSON files (either in the `pel_registry` module's installed path or in `/usr/share/phosphor-logging/pels/`), parse them, and create a nested dictionary like `componentsId[creatorId][componentId][componentName]`? This way, it can be used across all (or specific) PELs for any creator component IDs, reducing the need for repeated file lookups, which are currently impacting performance. I'm unsure how much improvement this will bring, which is why I requested performance data from #69.

Reply:
> I'm unsure how much improvement this will bring, which is why I requested performance data from https://github.com/ibm-openbmc/openpower-pel-parsers/pull/69.

It seems it will save less 4sec.

<img width="1062" alt="Screenshot 2024-10-17 at 10 11 27 AM" src="https://github.com/user-attachments/assets/44aca27f-6f41-449e-952a-6a57b0fd2e81">

Reply:
I believe the current logic unnecessarily repeats the module import in the non-BMC case, and we are attempting the import in the BMC case as well. It could be optimized to find the configuration file, parse it once, and reuse it.

For example
```
import sys

pelConfigRootPath = "/usr/share/phosphor-logging/pels"
attemptedToParseCompIDsCfg = False

def getAllCreatorsCompIDs():
    global attemptedToParseCompIDsCfg
    if attemptedToParseCompIDsCfg:
        return

    attemptedToParseCompIDsCfg = True
    componetsConfigPath = ""
    if os.path.exists(pelConfigRootPath): # BMC env
        componetsConfigPath = pelConfigRootPath
    else: # Non-BMC env
        try:
            import pel_registry
            componetsConfigPath = os.path.dirname(pel_registry.__file__)
        except ModuleNotFoundError:
            print("Unable to locate the PEL creators components configuration file", file=sys.stderr)
            return

    componentIDFileSuffix = "_component_ids.json"
    for file in os.listdir(componetsConfigPath):
        if componentIDFileSuffix not in file:
            continue
        with open(os.path.join(componetsConfigPath, file), 'r') as fileFd:
            creatorID = file[0:file.find(componentIDFileSuffix)]
            componentIDs[creatorID] = json.load(fileFd)
             
def getDisplayCompID(componentID: int, creatorID: str) -> str:
    // existing PHYP component id stuff

    if not componentIDs:
        getAllCreatorsCompIDs()

    compIDStr = '{:04X}'.format(componentID)
    if creatorID in componentIDs and compIDStr.upper() in componentIDs[creatorID]:
        return componentIDs[creatorID][compIDStr]

    return compIDStr
```